includes/compat/class-wc-stripe-subscriptions-helper.php

```diff
@@ -24,36 +36,65 @@ public static function is_subscriptions_enabled() {
 	}
 
 	/**
-	 * Loads up to 50 subscriptions, and attempts to return up to 5 of those that are detached from the customer.
+	 * Loads up to 50 subscriptions, and attempts to return those that are detached from the customer.
 	 *
 	 * @return array
 	 *
 	 * @deprecated 9.6.0 This method is no longer used and will be removed in a future version.
 	 */
 	public static function get_some_detached_subscriptions() {
 		_deprecated_function( __METHOD__, '9.6.0' );
+		return self::get_detached_subscriptions( 50 );
+	}
+
+	/**
+	 * Loads all active subscriptions renewing in less than a month, and attempts to return those that are detached from the customer.
+	 *
+	 * @param int $limit The maximum number of subscriptions to retrieve. Use -1 for no limit (default).
+	 * @return array
+	 */
+	public static function get_detached_subscriptions( $limit = -1 ) {
 		// Check if we have a cached result.
-		$cached_subscriptions = get_transient( self::DETACHED_SUBSCRIPTIONS_TRANSIENT_KEY );
-		if ( ! empty( $cached_subscriptions ) ) {
+		$cached_subscriptions = WC_Stripe_Database_Cache::get( self::DETACHED_SUBSCRIPTIONS_CACHE_KEY_PREFIX . $limit );
+		if ( is_array( $cached_subscriptions ) ) {
 			return $cached_subscriptions;
 		}
 
-		$subscriptions = wcs_get_subscriptions(
-			[
-				'subscriptions_per_page' => 50,
-				'page'                   => 1,
-				'orderby'                => 'date',
-				'order'                  => 'DESC',
-				'subscription_status'    => [ 'active', 'on-hold', 'pending-cancel' ],
-			]
-		);
+		$subscriptions = [];
+		$page          = 1;
+		$per_page      = self::MAX_SUBSCRIPTIONS_PER_PAGE;
+
+		do {
+			$batch             = wcs_get_subscriptions(
+				[
+					'subscriptions_per_page' => $per_page,
+					'page'                   => $page,
```

Comment:
```suggestion
					'paged'                   => $page,
```

The argument name should be `paged` ([ref](https://github.com/woocommerce/woocommerce-subscriptions/blob/19b8edeff75b558c15ebc8aa9f285736e11573f1/includes/core/wcs-functions.php#L420C4-L420C9))

I discovered it as I changed the `MAX_SUBSCRIPTIONS_PER_PAGE` value to `10`. This caused a fatal error in my site.

<img width="1306" alt="Screenshot 2025-06-23 at 2 01 45 PM" src="https://github.com/user-attachments/assets/bd9739d2-0c14-4454-a82d-da617668e966" />

Reply:
Nice catch, thanks! Suggestion applied ✅